api/app/utils/parse_verb_string.py

```diff
@@ -0,0 +1,20 @@
+def parse_verb_string(s):
```

Comment:
I would write in this manner 

def parse_verb_string(s):
    parts = s.split('+')

    if len(parts) == 4:
        negative, subject, tense, root = parts
    elif len(parts) == 3:
        subject, tense, root = parts
        negative = ''
    else:
        raise ValueError('Invalid string format')

    # Check if root is a compound verb: Kuwa na
    root, *compound_verb = root.split()
    compound_verb = ' ' + compound_verb[0] if compound_verb else ''

    negative = negative if negative == 'NEG' else ''

    return negative, subject, tense, root, compound_verb